Ferrite 1.0 compat

termi-official opened the pull request:
In this branch and PR I develop the preparation for the upcoming Ferrite release.

## TODOs

* [x] Replace `transfer_quadrature_face_to_cell` with the Ferrite internal one
* [x] Remove `getgrid`

## Post-Merge

* [ ] Enable reverse CI https://github.com/Ferrite-FEM/Ferrite.jl/blob/ffab3f0c64707d0dd5439ab1dc94c3b0927a76b2/.github/workflows/Downstream.yml#L29

termi-official committed:
First try to upgrade to Ferrite master

termi-official committed:
Update atopics.

termi-official commented on src/utils.jl:
```diff
@@ -1,25 +1,38 @@
+    c = c0 + 1
+    v = Ferrite.shape_value(ipv.ip, ξ, i)
+    return SVector(ntuple(j -> j == c ? v : zero(v), 4))
 end
 
-function field_offset(dh::Ferrite.DofHandler, field_idx::Int)
-    offset = 0
-    for i in 1:(field_idx-1)
-        offset += Ferrite.getnbasefunctions(dh.field_interpolations[i])::Int * dh.field_dims[i]
-    end
-    return offset
+function Ferrite.shape_value(ipv::VectorizedInterpolation{9, shape}, ξ::Tensors.Vec{3, T}, I::Int) where {shape <: Ferrite.AbstractRefShape{3}, T}
+    i0, c0 = divrem(I - 1, 9)
+    i = i0 + 1
+    c = c0 + 1
+    v = Ferrite.shape_value(ipv.ip, ξ, i)
+    return SVector(ntuple(j -> j == c ? v : zero(v), 9))
 end
```
xref https://github.com/Ferrite-FEM/Ferrite.jl/issues/398

termi-official committed:
Make gradient fields native.

termi-official committed:
Broken manufactured linear elastic example + first take on testing vector valued tests to track down the bug in triangle gradients.

termi-official commented on docs/src/ferrite-examples/linear-elasticity.jl:
```diff
@@ -0,0 +1,107 @@
+        G = E/2(1+ν)
+        K = E/3(1-2ν)
+
+        u(x)  = Vec{2}(i->i==component ? (π/2)^2 * 2 * prod(cos, x*π/2) : 0.0)
+        ∇u(x) = Tensors.gradient(u, x)
+        ε(x)  = (∇u(x)  + transpose(∇u(x)))/2
+        σ(x)  = 2G * dev(ε(x)) + K * tr(ε(x)) * one(ε(x))
```
I guess there is an issue with the symmetry here?

termi-official commented on test/runtests.jl:
```diff
@@ -72,4 +73,73 @@ include("../docs/src/ferrite-examples/heat-equation.jl")
+            end
+        end
+    end
+end
```
I guess we should split this and move the script into https://github.com/Ferrite-FEM/Ferrite.jl/pull/640 and use `apply_analytical` and a point-wise evaluation of the gradients here.

termi-official committed:
Fix gradient interpolation+regression test.

termi-official committed:
Fix documenter build step 1.

termi-official committed:
Images now show but build errors because the pages are too big :)

termi-official committed:
Cleanup.

termi-official committed:
Merge master.

termi-official committed:
Temporary master CI. Co-authored by @KnutAM

termi-official committed:
Update quadrature transfer with the Ferrite version

termi-official committed:
Fix doc CI.

termi-official committed:
Drop support for old Makie versions.

termi-official committed:
Fix shading warning.

termi-official committed:
Disable cache temporarily.

termi-official commented on .github/workflows/docs.yml:
```diff
@@ -13,14 +13,14 @@ jobs:
-            ~/.julia/registries
-          key: .julia-docs-${{ hashFiles('docs/Project.toml', 'docs/Manifest.toml') }}
+          version: '1.10'
+      # - uses: actions/cache@v3
+      #   with:
+      #     path: |
+      #       ~/.julia/artifacts
+      #       ~/.julia/packages
+      #       ~/.julia/registries
+      #     key: .julia-docs-${{ hashFiles('docs/Project.toml', 'docs/Manifest.toml') }}
```
Before merge
```suggestion
      - uses: actions/cache@v3
        with:
          path: |
            ~/.julia/artifacts
            ~/.julia/packages
            ~/.julia/registries
          key: .julia-docs-${{ hashFiles('docs/Project.toml', 'docs/Manifest.toml') }}
```

KnutAM commented:
Found today that `elementinfo(::Interpolation)` is broken on this branch.

termi-official committed:
Update to new Ferrite and Makie versions. Most 3D plots are now broken for some reason when there is also a 2D version of it.

termi-official commented:
Reminder for myself - There is some kind of regression in the Makie API behavior which I need to investigate. MWE:
```
using FerriteViz.Makie, WGLMakie, Bonito

Page()

struct MakiePlotter{dim}
end

@recipe(MyLines) do scene
    Attributes(
        plotnodes=true,
    )
end


function Makie.plot!(WF::MyLines{<:Tuple{<:MakiePlotter{dim}}}) where dim
    plotter = WF[1][]
    if dim == 2
        lines = [Point2f(0.0,0.0)]
    else
        lines = [Point3f(1.0,1.0,0.0), Point3f(0.0,0.0,1.0)]
        @info "Peekaboo"
    end
    Makie.scatter!(WF, lines)
end

mylines(MakiePlotter{2}())  # 2D
mylines(MakiePlotter{3}())  # 2D
scatter([Point2f(0.0,0.0)]) # 2D
scatter([Point3f(1.0,1.0,0.0), Point3f(0.0,0.0,1.0)]) # 3D
```

termi-official committed:
Manifest.

termi-official committed:
Fix axis query.

termi-official committed:
Bump Ferrite version.

termi-official committed:
Fix deprecation warnings

termi-official committed:
Update to new master

termi-official committed:
Merge master

termi-official committed:
Fix axis dispatches

termi-official committed:
Derp

termi-official committed:
Derp again

termi-official committed:
Update CI. Ferrite does not work with 1.6 anymore.

termi-official committed:
More updates

termi-official committed:
Try to fix docs CI

termi-official committed:
Fix last hiccups on release version

termi-official commented:
There is some issue regarding the depth with the interactivity. Other than that good from my side.

fredrikekre committed:
Use the grids geometric interpolation in transfer solution

termi-official committed:
Try to fix docs CI

koehlerson commented:
> There is some issue regarding the depth with the interactivity. Other than that good from my side. @koehlerson if you agree with the changes we can merge and bump.

Anything in particular? Will look into that this week

termi-official commented:
> > There is some issue regarding the depth with the interactivity. Other than that good from my side. @koehlerson if you agree with the changes we can merge and bump.
> 
> Anything in particular? Will look into that this week

CI changes and some changes in utils.jl (e.g. I removed ε as it caused issues). Everything else should be a very mechanistic update.

termi-official committed:
Update to new Makie version

termi-official committed:
Cleanup api docs.

termi-official committed:
Fix elementinfo

termi-official commented:
> elementinf

Okay I need to fix this first.

koehlerson commented:
while checking if everything interactive works, I stumbled across this problem: https://github.com/Ferrite-FEM/FerriteViz.jl/issues/131

termi-official committed:
Pin to old MakieCore version until time to change the FerriteViz API comes.

termi-official committed:
Compat pls?

termi-official committed:
Compat pls?

termi-official commented:
If CI passes I would suggest that we simply bump the version for a new release and fix the update issue in a subsequent PR. What do you think @koehlerson ?

termi-official committed:
Remove unused packages

termi-official committed:
Update interpolation type

termi-official committed:
is_discontinous breakage

koehlerson committed:
rename of plot recipes (#140)

This temporarily changes the name of the recipes. This is the easiest way to ensure compat with the current Makie release.

termi-official committed:
Merge master

termi-official committed:
Docs cache version

koehlerson commented:
<img width="888" height="499" alt="grafik" src="https://github.com/user-attachments/assets/752df5f0-96e9-4630-b6cc-bd563b19ac66" />

termi-official committed:
Just one more hack
